### ISSUE TITLE
`NUnit2045` does not respect lambda captures

### ISSUE THREAD

**Issue by RenderMichael** [nunit/nunit.analyzers#614]
I have a test like this

``` csharp
[Test]
public void MyTest()
{
    object? actualDeserialized = null;

    Assert.That(() => actualDeserialized = Calculate(), Throws.Nothing);
    Assert.That(actualDeserialized, Is.Not.Null);
}

private static object? Calculate() { return new(); }
```

This triggers `NUnit2045`. However, because the lambda captures the `actualDeserialized` variable, these two assertions are not independent.

Is there any way Roslyn can delect when a lambda captures a certain variable, so `NUnit2045` can be disabled for this circumstance?

edit: my minimal repro did demonstrate the issue, but was a little nonsensical. I tweaked it.
